LoopKitUI/Views/InsulinSensitivityScheduleEditor.swift

```diff
@@ -0,0 +1,115 @@
+    )
+}
+
+public struct InsulinSensitivityScheduleEditor: View {
+    private var schedule: DailyQuantitySchedule<Double>?
+    private var glucoseUnit: HKUnit
+    private var save: (InsulinSensitivitySchedule) -> Void
+
+    public init(
+        schedule: InsulinSensitivitySchedule?,
+        glucoseUnit: HKUnit,
+        onSave save: @escaping (InsulinSensitivitySchedule) -> Void
+    ) {
+        // InsulinSensitivitySchedule stores only the glucose unit.
+        // For consistency across display & computation, convert to "real" <glucose unit>/U units.
+        self.schedule = schedule.map { schedule in
+            DailyQuantitySchedule(
+                unit: glucoseUnit.unitDivided(by: .internationalUnit()),
+                dailyItems: schedule.items
+            )!
+        }
+        self.glucoseUnit = glucoseUnit
+        self.save = save
+    }
+
+    public var body: some View {
+        QuantityScheduleEditor(
+            title: Text("Insulin Sensitivities", comment: "Title of insulin sensitivity settings page"),
+            description: description,
+            schedule: schedule,
+            unit: sensitivityUnit,
+            guardrail: .insulinSensitivity,
+            selectableValueStride: stride,
+            defaultFirstScheduleItemValue: Guardrail.insulinSensitivity.absoluteBounds.upperBound,
+            confirmationAlertContent: confirmationAlertContent,
+            guardrailWarning: InsulinSensitivityGuardrailWarning.init(crossedThresholds:),
+            onSave: {
+                // Convert back to the expected glucose-unit-only schedule.
+                self.save(DailyQuantitySchedule(unit: self.glucoseUnit, dailyItems: $0.items)!)
+            }
+        )
+    }
+
+    private var description: Text {
+        Text("Your insulin sensitivity factor (ISF) is the drop in glucose expected from one unit of insulin.", comment: "Insulin sensitivity setting description")
+    }
+
+    private var sensitivityUnit: HKUnit {
+         glucoseUnit.unitDivided(by: .internationalUnit())
+    }
+
+    private var stride: HKQuantity {
+        let doubleValue: Double
+        switch glucoseUnit {
+        case .milligramsPerDeciliter:
+            doubleValue = 1
+        case .millimolesPerLiter:
+            doubleValue = 0.1
+        case let otherUnit:
+            fatalError("Unsupported glucose unit \(otherUnit)")
+        }
+
+        return HKQuantity(unit: sensitivityUnit, doubleValue: doubleValue)
+    }
+
+    private var confirmationAlertContent: AlertContent {
+        AlertContent(
+            title: Text("Save Insulin Sensitivities?", comment: "Alert title for confirming insulin sensitivities outside the recommended range"),
+            message: Text("Some of the values you have entered are outside of what Tidepool generally recommends.", comment: "Alert message for confirming insulin sensitivities outside the recommended range")
```

Comment:
Did we change this text to be "what *we* recommend" in a different view? Just thinking of DIY here.

Reply:
We didn't, unfortunately; it sounded from a conversation between Howard, Melissa, and Matt that Tidepool owning responsibility for these guardrails is important. It does mean we'll need some new replaceable string facility to integrate with DIY.

(Suspend threshold has the same problem, though it's perhaps even trickier here since this string exists at the framework level.)